### ISSUE TITLE
unused or repeated assign to the same index

### ISSUE THREAD

**Issue by bingmatv** [rust-lang/rust-clippy#13908]
### What it does

To detect whether the index is the same and whether there's unused or repeated assign.

### Example

```rust
fn main() {
    let mut a = [0u8; 8];
    a[0] = 1;
    a[0] = 1;
    a[0..3].copy_from_slice(&[1; 3]);
}
```
Clippy didn't warn there are repeated assign to the index 0.
